hikari/internal/deprecation.py

```diff
@@ -35,72 +35,63 @@
     T = typing.TypeVar("T", bound=typing.Callable[..., typing.Any])
 
 
-def warn_deprecated(
-    obj: typing.Any,
-    /,
-    *,
-    version: typing.Optional[str] = None,
-    alternative: typing.Optional[str] = None,
-    stack_level: int = 3,
-) -> None:
+def warn_deprecated(obj: typing.Any, additional_information: str, /, *, stack_level: int = 3) -> None:
     """Raise a deprecated warning.
 
     Parameters
     ----------
     obj: typing.Any
         The object that is deprecated.
+    additional_information: str
+        Additional information on the deprecation for the user.
 
     Other Parameters
     ----------------
-    version: typing.Optional[str]
-        If specified, the version it will be removed in.
-    alternative: typing.Optional[str]
-        If specified, the alternative to use.
     stack_level: int
         The stack level for the warning. Defaults to `3`.
     """
-    if inspect.isclass(obj) or inspect.isfunction(obj):
+    if inspect.isclass(obj):
+        action = ("Instantiation of", "class")
```

Comment:
Also this isn't fit for classes as it'd break anything inheriting from said class so this seems a bit redundant